azure/functions/decorators/function_app.py

```diff
@@ -3297,8 +3287,27 @@ def get_functions(self) -> List[Function]:
                 '-bindings-http-webhook-trigger?tabs=in-process'
                 '%2Cfunctionsv2&pivots=programming-language-python#http-auth')
 
+        FunctionRegister.validate_functions(self, functions=functions)
+
         return functions
 
+    def validate_functions(self, functions: List[Function]) -> bool:
+        """The functions_bindings dict contains the function name and
+        its bindings for all functions in an app. If a previous function
+        has the same name, indexing will fail here.
+        """
+        if not self.functions_bindings:
+            self.functions_bindings = {}
+        for function in functions:
+            function_name = function.get_function_name()
+            if self.functions_bindings.get(function_name, None):
+                raise ValueError(
+                    f"Function {function_name} does not have a unique"
+                    f" function name. Please change @app.function_name() or"
+                    f" the function method name to be unique.")
+            self.functions_bindings[function_name] = function.get_bindings()
```

Comment:
Why another call to get_bindings if we only need the function name?

Reply:
The value of the key-value pair in the dict right now is the corresponding function bindings. It's an O(1) call since it's just returning the corresponding variable. I thought it might be helpful for future records / debugging

Reply:
we can keep the dictionary as
{
    function_name1: 1
    function_name2: 1
}

since we are only interested in the keys the values won't matter right?
We avoid another method calling hop in that way